lstore/transaction_worker.py

```diff
@@ -1,44 +1,80 @@
 from lstore.table import Table, Record
 from lstore.index import Index
+from lstore.lock import LockManager
+import threading
+import logging
+import time
 
 class TransactionWorker:
-
+    
     """
     # Creates a transaction worker object.
     """
-    def __init__(self, transactions = []):
+    def __init__(self, transactions = None, lock_manager = None, deadlock_detection = True):
         self.stats = []
-        self.transactions = transactions
+        self.transactions = transactions if transactions is not None else []
         self.result = 0
-        pass
+        self.worker_thread = None
+        self.lock_manager = lock_manager if lock_manager is not None else LockManager()        
+        self.lock = threading.Lock()
+        self.transaction_errors = {}
+        self.deadlock_detection = deadlock_detection
+        self.timeout = 100        # Some large number for now since idk what to set it to
+
 
-    
     """
     Appends t to transactions
     """
     def add_transaction(self, t):
-        self.transactions.append(t)
-
-        
+        with self.lock:
+            self.transactions.append(t)
+    
     """
     Runs all transaction as a thread
     """
     def run(self):
-        pass
-        # here you need to create a thread and call __run
+        self.worker_thread = threading.Thread(target = self.__run)
+        self.worker_thread.start()
     
-
     """
     Waits for the worker to finish
     """
     def join(self):
-        pass
+        if self.worker_thread is not None:
+            self.worker_thread.join()
+        return self.result
+    
+    def __run(self):
+        transactions_to_run = []
+        
+        with self.lock:
+            transactions_to_run = list(self.transactions)
+        
+        for transaction in transactions_to_run:
+            transaction_id = id(transaction)
+            start_time = time.time()
+            result = None
+            
+            try:
 
+                self.lock_manager.transaction_states[transaction_id] = 'growing'                
```

Comment:
all work after this line should be part of lock_manager, which automatically adds the growing phase to each transaction. Also the deadlock detection would be a part of lock manager and they way I'm thinking rn is it will give you an exception if a deadlock is deetected when grabbing a lock for a transaction